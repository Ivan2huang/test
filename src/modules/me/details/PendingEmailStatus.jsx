import React from 'react';
import PropTypes from 'prop-types';
import { injectIntl } from 'react-intl';

import { withStyles } from '@material-ui/core/styles';
import NotificationBox from '../../../uiComponents/NotificationBox';
import Typography from '../../../uiComponents/Typography';
import WarningLg from '../../../icons/WarningLg';

import { formatMessage } from '../../../helpers/helpers';

const styles = theme => ({
  warningIcon: {
    marginRight: theme.spacing(1),
  },
});

const PendingEmailStatus = ({ classes, intl }) => {
  return (
    <NotificationBox>
      <WarningLg className={classes.warningIcon} />
      <Typography type="style-6">
        {formatMessage(
          intl,
          'me.details.pendingEmailRequest.label',
          'Pending email address change request.',
        )}
      </Typography>
    </NotificationBox>
  );
};

PendingEmailStatus.propTypes = {
  classes: PropTypes.shape({}).isRequired,
  intl: PropTypes.shape({}).isRequired,
};

export default withStyles(styles)(injectIntl(PendingEmailStatus));
