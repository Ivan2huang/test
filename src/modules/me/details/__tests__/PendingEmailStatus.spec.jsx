import React from 'react';
import { render } from '@testing-library/react';
import PendingEmailStatus from '../PendingEmailStatus';
import withTheme from '../../../../themes/withThemeProvider';
import withIntl from '../../../../i18n/withIntlProvider';

jest.mock(
  '../../../../uiComponents/Typography',
  // eslint-disable-next-line react/prop-types
  () => ({ children, ...rest }) => <div {...rest}>{children}</div>,
);

describe('PendingEmailStatus Component', () => {
  it('should match the snapshot', () => {
    const Component = withIntl(withTheme(PendingEmailStatus));
    const { container } = render(<Component />);
    expect(container).toMatchSnapshot();
  });
});
