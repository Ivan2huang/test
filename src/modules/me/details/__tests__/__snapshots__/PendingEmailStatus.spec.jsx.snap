// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PendingEmailStatus Component should match the snapshot 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-6 makeStyles-container-3 makeStyles-container-4 NotificationBox-container-2 "
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root injectIntl(PendingEmailStatus)-warningIcon-1"
      focusable="false"
      role="presentation"
      style="width: 1.125rem; height: 1.125rem;"
      viewBox="0 0 18 18"
    >
      <path
        d="M9 18C13.9706 18 18 13.9706 18 9C18 4.02944 13.9706 0 9 0C4.02944 0 0 4.02944 0 9C0 13.9706 4.02944 18 9 18Z"
        fill="#FFBB33"
      />
      <path
        d="M9.00098 6.25C9.69133 6.25 10.251 5.69036 10.251 5C10.251 4.30964 9.69133 3.75 9.00098 3.75C8.31062 3.75 7.75098 4.30964 7.75098 5C7.75098 5.69036 8.31062 6.25 9.00098 6.25Z"
        fill="#333333"
      />
      <path
        d="M9.89997 8.25H8.09998V14.25H9.89997V8.25Z"
        fill="#333333"
      />
    </svg>
    <div
      type="style-6"
    >
      Pending email address change request.
    </div>
  </div>
</div>
`;
